otter/tap/api.py

```diff
@@ -278,17 +279,22 @@ def on_client_ready(_):
                                              get_service_configs(config),
                                              kz_client, store, supervisor,
                                              cassandra_cluster)
+
             # Setup scheduler service after starting
             scheduler = setup_scheduler(parent, dispatcher, store, kz_client)
-            health_checker.checks['scheduler'] = scheduler.health_check
-            otter.scheduler = scheduler
+            if scheduler is not None:
```

Comment:
> I don't see anywhere that setup_scheduler itself has been modified to return a different value.

https://github.com/rackerlabs/otter/pull/1887/files#diff-17282da832855fb34e8e70ff0ec5ddd1R349. This is just explicit. Actually, `setup_scheduler` had always returned `None` when it didn't find scheduler config but that case was never handled by the caller (this code). This code now handles that case.

> If it really needs to return a new type, could you consider an Active Nothing here?

Sorry what is `Active Nothing`. To be clear, `setup_scheduler` returns scheduler service object based on the config. If config is not found, it returns `None`. I'll add this to the docstring.